src/mcp/tools/dataconnect/generate_dataconnect_operation.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { toContent } from "../../util.js";
+import { generateOperation } from "../../../gif/fdcExperience.js";
+import { Config } from "../../../config.js";
+import { pickService } from "../../../dataconnect/fileUtils.js";
+
+export const generate_dataconnect_operation = tool(
+  {
+    name: "generate_dataconnect_operation",
+    description:
+      "Generates a single Firebase Data Connect query or mutation based on the currently deployed schema and the provided prompt.",
+    inputSchema: z.object({
+      // Lifted guidance from : https://cloud.google.com/gemini/docs/discover/write-prompts
+      prompt: z
+        .string()
+        .describe(
+          "Write the prompt like you're talking to a person, describe the task you're trying to accomplish and give details that are specific to the users requst",
+        ),
+      service: z
+        .string()
+        .nullish()
+        .describe(
+          "Optional: Uses the service from the firebase.json file if nothing provided. The service id or name of the deployed Firebase Data Connect Schema in format: projects/<project-id>/locations/<location>/services/<service-name>.",
+        ),
+    }),
+    annotations: {
+      title:
+        "Generate a Firebase Data Connect Operation on a deployed Firebase Data Connect Schema.",
+      readOnlyHint: true,
+    },
+    _meta: {
+      requiresProject: true,
+      requiresAuth: true,
+      // TODO: Create an endpoint to check for GiF activiation.
+    },
+  },
+  async ({ prompt, service }, { projectId, host }) => {
+    const serviceInfo = await pickService(
+      projectId!,
+      Config.load({ cwd: host.projectRoot })!,
```

Comment:
nit: Just added `config` as a context passed to all tools.

We can re-use that

Reply:
Done!